src/main/java/org/prebid/server/bidder/thetradedesk/TheTradeDeskBidder.java

```diff
@@ -0,0 +1,206 @@
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.iab.openrtb.request.App;
+import com.iab.openrtb.request.Banner;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Format;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Publisher;
+import com.iab.openrtb.request.Site;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.MultiMap;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.thetradedesk.ExtImpTheTradeDesk;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.BidderUtil;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+public class TheTradeDeskBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpTheTradeDesk>> TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private static final String PREBID_INTEGRATION_TYPE_HEADER = "x-integration-type";
+    private static final String PREBID_INTEGRATION_TYPE = "1";
+    private static final String SUPPLY_ID_MACRO = "{{SupplyId}}";
+    private static final Pattern SUPPLY_ID_PATTERN = Pattern.compile("([a-z]+)$");
+
+    private final String endpointUrl;
+    private final String supplyId;
+    private final JacksonMapper mapper;
+
+    public TheTradeDeskBidder(String endpointUrl, JacksonMapper mapper, String supplyId) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.supplyId = validateSupplyId(supplyId);
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    private static String validateSupplyId(String supplyId) {
+        if (StringUtils.isBlank(supplyId) || SUPPLY_ID_PATTERN.matcher(supplyId).matches()) {
+            return supplyId;
+        }
+
+        throw new IllegalArgumentException("SupplyId must be a simple string provided by TheTradeDesk");
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<Imp> modifiedImps = new ArrayList<>();
+
+        String publisherId = StringUtils.EMPTY;
```

Comment:
They are using an empty string because nil cannot be used as an assignment for the string type in Go. Update it to the simpler Java `null`. Then, on line 76, a simple `!= null` check will be sufficient.